cloud/pkg/controllermanager/edgeapplication/overridemanager/commandoverrider.go

```diff
@@ -0,0 +1,144 @@
+	}
+	return nil, nil
+}
+func buildCommandArgsPatchesWithPath(target string, specContainersPath string, rawObj *unstructured.Unstructured, commandRunOverrider *v1alpha1.CommandArgsOverrider) ([]overrideOption, error) {
+	patches := make([]overrideOption, 0)
+	containers, ok, err := unstructured.NestedSlice(rawObj.Object, strings.Split(specContainersPath, pathSplit)...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieves path(%s) from rawObj, error: %v", specContainersPath, err)
+	}
+	if !ok || len(containers) == 0 {
+		return nil, nil
+	}
+	klog.V(4).Infof("buildCommandArgsPatchesWithPath containers info (%+v)", containers)
+	for index, container := range containers {
+		if container.(map[string]interface{})["name"] == commandRunOverrider.ContainerName {
+			commandArgsPath := fmt.Sprintf("/%s/%d/%s", specContainersPath, index, target)
+			commandArgsValue := make([]string, 0)
+			var patch overrideOption
+			// if target is nil, to add new [target]
+			if container.(map[string]interface{})[target] == nil {
+				patch, _ = acquireAddOverrideOption(commandArgsPath, commandRunOverrider)
+			} else {
+				for _, val := range container.(map[string]interface{})[target].([]interface{}) {
+					commandArgsValue = append(commandArgsValue, fmt.Sprintf("%s", val))
+				}
+				patch, _ = acquireReplaceOverrideOption(commandArgsPath, commandArgsValue, commandRunOverrider)
+			}
+
+			klog.V(4).Infof("[buildCommandArgsPatchesWithPath] containers patch info (%+v)", patch)
+			patches = append(patches, patch)
+		}
+	}
+	return patches, nil
+}
+
+func acquireAddOverrideOption(commandArgsPath string, commandOverrider *v1alpha1.CommandArgsOverrider) (overrideOption, error) {
+	if !strings.HasPrefix(commandArgsPath, pathSplit) {
+		return overrideOption{}, fmt.Errorf("internal error: [acquireCommandOverrideOption] commandRunPath should be start with / character")
+	}
+	newCommandArgs, err := overrideCommandArgs([]string{}, commandOverrider)
+	if err != nil {
+		return overrideOption{}, err
+	}
+	return overrideOption{
+		Op:    string(v1alpha1.OverriderOpAdd),
+		Path:  commandArgsPath,
+		Value: newCommandArgs,
+	}, nil
+}
+
+func acquireReplaceOverrideOption(commandArgsPath string, commandArgsValue []string, commandOverrider *v1alpha1.CommandArgsOverrider) (overrideOption, error) {
+	if !strings.HasPrefix(commandArgsPath, pathSplit) {
+		return overrideOption{}, fmt.Errorf("internal error: [acquireCommandOverrideOption] commandRunPath should be start with / character")
+	}
+
+	newCommandArgs, err := overrideCommandArgs(commandArgsValue, commandOverrider)
+	if err != nil {
+		return overrideOption{}, err
+	}
+
+	return overrideOption{
+		Op:    string(v1alpha1.OverriderOpReplace),
+		Path:  commandArgsPath,
+		Value: newCommandArgs,
+	}, nil
+}
+
+func overrideCommandArgs(curCommandArgs []string, commandArgsOverrider *v1alpha1.CommandArgsOverrider) ([]string, error) {
```

Comment:
You don't have to return an error here, you can also change the return value of the call.

Reply:
Yes, I have made changes